### PR TITLE
"To Do" - project: Week 15 - Ida Sylvén

### PR DESCRIPTION
**Pull request by idasylven**
Link to my deployed project:

https://reverent-engelbart-256280.netlify.app/

My reflections on how this week's project turned out:

Once again I get more confident when we go through the core structure thoroughly during the lectures. I really learn way better from implementing my own codealongs and getting used to typing the syntax. Im really fond of the idea of the store, and understand how useful it is to work with - I just need to keep practicing and get more used to the flow and interaction between the functionalities. I aimed for Blue Level requirements, and proceeded using 'styled-components' instead of vanilla CSS as part of one of the Red Level stretch goals.  

Things I'd like to have clarified or explained in more detail:

A lot of things I guess, in terms of getting into understanding the slices/reducers/actions a bit better - but generally I think I just need to keep practicing!

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -4,6 +4,13 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link href="https://fonts.googleapis.com/css2?family=Reenie+Beanie&display=swap" rel="stylesheet">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@700&display=swap" rel="stylesheet">    
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Rock+Salt&display=swap" rel="stylesheet">
+    <link rel="icon" href="https://emojipedia-us.s3.dualstack.us-west-1.amazonaws.com/thumbs/120/apple/271/memo_1f4dd.png">
+
```
Comment (IsabellaM5) [inline code/public/index.html:13]: Loved the fonts! 🤩

File: code/src/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  font-size: 2.5rem;
+  font-family: Rock Salt;
+
+  mark {
+    font-weight: bolder;
+    background: linear-gradient(104deg, rgba(130, 255, 173,0) 0.9%, rgba(130, 255, 173,1.25) 2.4%, rgba(130, 255, 173,0.5) 5.8%, rgba(130, 255, 173,0.1) 93%, rgba(130, 255, 173,0.7) 96%, rgba(130, 255, 1732,0) 98%), linear-gradient(183deg, rgba(130, 255, 173,0) 0%, rgba(130, 255, 173,0.3) 7.9%, rgba(130, 255, 173,0) 15%);
+    border-radius: 7.5px;
+    }
+    @media (min-width: 1200px) {
+    margin: 0 30rem 0 30rem;
```
Comment (IsabellaM5) [inline code/src/components/Header.js:24]: When on breakpoint 1200px, the content spills over the sides in both Header, TodoForm and TodoList (see picture). I would consider checking the margin applied here and reduce it a bit. 🙂

image

File: code/src/components/TodoForm.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      isComplete: false
+    }
+
+    dispatch(todos.actions.addTodo(newTodo))
+    setValue('')
+  }
+
+  return (
+    <AddContainer>
+      <form onSubmit={onFormSubmit} spellCheck="false">
```
Comment (IsabellaM5) [inline code/src/components/TodoForm.js:63]: One suggestion is to have a styled component for this form element as well 🙂

File: code/src/components/TodoList.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+const PostIt = styled.div`
+  width: 17rem;
+  height: 7rem;
+  background: #e8d4e3;
+  margin: 2rem;
+  box-shadow: 5px 5px 7px rgb(33 33 33 / 39%);
+  display: flex;
+  justify-content: space-between;
+
+  [type=checkbox] {
```
Comment (IsabellaM5) [inline code/src/components/TodoList.js:27]: Great styling here of the checkboxes! I’ll have to look this up myself for future projects 🙂✨

File: code/src/components/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+           <input 
+            type="checkbox"
+            checked={todo.isComplete}
+            onChange={() => dispatch(todos.actions.toggleComplete(todo.id))}
+          />
+            <NoteText>{todo.description}</NoteText>
+              <BinButton onClick={() => dispatch(todos.actions.removeTodo(todo.id))}>
+                <ImBin2 />
+              </BinButton>
+        </PostIt>
```
Comment (IsabellaM5) [inline code/src/components/TodoList.js:84]: As you’ve already pointed out in the README file, these can be made into smaller components coupled with their styled components to reduce the amount of code in this file. 🧐

File: code/src/index.css [hunk 6]
```diff
@@ -11,3 +11,9 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
+@media (min-width: 1200px) {
+  body {
+    background-color: rgba(245, 245, 236, 0.39);
+  }
```
Comment (IsabellaM5) [inline code/src/index.css:18]: One suggestion is to make use of the createGlobalStyle feature in styled components. You can read about it here in the docs (just search for global styles): https://styled-components.com/docs/api 🤓

**Review by IsabellaM5 — APPROVED** [review]
Such a fantastic job with this project Ida! 😄 🐯

I really loved your idea with this project and your styling truly incorporates the feeling of a To Do application! It made me want to pick up a pen and start writing todos right away! ✍️

You got the basic functionality in place in accordance with the blue level requirements. You also stretched your muscles and did styling with styled components! 💅 Great job! 💪 Also, you have some great future ambitions for this project - this is definitely a project where you can continue to add things to. Keep at it! Practice makes perfect 🙌 ✨

Your project is hereby approved! 👍 For some specific comments/suggestions, see the comments made in the code files.

/Isabella 🐯

File: code/src/components/TodoForm.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        <AddButton type="submit">+
+        </AddButton>
+        <InputField  
+          type="text"
+          placeholder="Add POST IT" 
+          value={value}
+          onChange={event => setValue(event.target.value)}>
```
Comment (KarolCislowski) [inline code/src/components/TodoForm.js:70]: FIGHT WITH LAZYNESS!! 🔫 add REQUIRED to make your to-do free from empty tasks!! 🎖️ 😆

**Review by KarolCislowski — APPROVED** [review]
Great idea with post-it convention 🌈

**Pull request closed without merging**